### PR TITLE
feat(`utils`): Add primitive types conversion traits

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

It's extremely painful to have to write `u256_to_ru256(stuff)` when we could have a trait for this.

## Solution

Add conversion traits for:
- Alloy -> Ethers
- Ethers -> Alloy

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
lgmt, pending @DaniPopes

File: crates/utils/src/types.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    fn to_alloy(self) -> Self::To {
+        Address::from_slice(self.as_bytes())
+    }
+}
+
+impl ToAlloy for H256 {
+    type To = alloy_primitives::B256;
+
+    fn to_alloy(self) -> Self::To {
+        B256::from_slice(self.as_bytes())
```
Comment (DaniPopes) [inline crates/utils/src/types.rs:26]: B<->H are the exact same, using slices re allocates so use `B256::new(self.0)`

File: crates/utils/src/types.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+}
+
+impl ToAlloy for U256 {
+    type To = alloy_primitives::U256;
+
+    fn to_alloy(self) -> Self::To {
+        let mut buffer = [0u8; 32];
+        self.to_little_endian(buffer.as_mut_slice());
+        AlloyU256::from_le_bytes(buffer)
```
Comment (DaniPopes) [inline crates/utils/src/types.rs:36]: afaik ruint is the same as well, same as B types

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/utils/src/types.rs (hunk 2 above)
Comment (Evalir) [reply crates/utils/src/types.rs]: done in https://github.com/foundry-rs/foundry/pull/5846/commits/70025ea32077835a0265a34fd5ff81b952ae1992 señor

File: crates/utils/src/types.rs (hunk 1 above)
Comment (Evalir) [reply crates/utils/src/types.rs]: done in https://github.com/foundry-rs/foundry/pull/5846/commits/70025ea32077835a0265a34fd5ff81b952ae1992

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**